File: backend/chat-store-service/src/main/java/com/example/chatstoreservice/common/util/TimeUtil.java
package com.example.chatstoreservice.common.util;

import lombok.extern.slf4j.Slf4j;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.TimeZone;

@Slf4j
public final class TimeUtil {

    public static LocalDateTime converDateToLocalDateTime(Long date) {
        return date == null ? LocalDateTime.now() : LocalDateTime.ofInstant(Instant.ofEpochMilli(date), TimeZone.getDefault().toZoneId());
    }

    public static LocalDateTime converDateToLocalDateTime(Date date) {
        return date == null ? LocalDateTime.now() : date.toInstant().atZone(TimeZone.getDefault().toZoneId()).toLocalDateTime();
    }
}

File: backend/chat-store-service/src/main/kotlin/com/example/chatstoreservice/common/util/TimeUtil.kt
package com.example.chatstoreservice.common.util

import java.time.Instant
import java.time.LocalDateTime
import java.util.Date
import java.util.TimeZone

object TimeUtil {

    @JvmStatic
    fun converDateToLocalDateTime(date: Long?): LocalDateTime =
        if (date == null) LocalDateTime.now() else LocalDateTime.ofInstant(Instant.ofEpochMilli(date), TimeZone.getDefault().toZoneId())

    @JvmStatic
    fun converDateToLocalDateTime(date: Date?): LocalDateTime =
        date?.toInstant()?.atZone(TimeZone.getDefault().toZoneId())?.toLocalDateTime() ?: LocalDateTime.now()
}
